pkgs/development/libraries/qt-6/modules/qtwebengine.nix

```diff
@@ -77,33 +97,69 @@ qtModule rec {
     flex
     git
     gperf
-    ninja
+    ninja-tokenpool
     pkg-config
     (python3.withPackages (ps: with ps; [ html5lib ]))
     which
-    gn
     nodejs
   ];
   doCheck = true;
   outputs = [ "out" "dev" ];
 
-  dontUseGnConfigure = true;
-
   # ninja builds some components with -Wno-format,
   # which cannot be set at the same time as -Wformat-security
   hardeningDisable = [ "format" ];
 
   patches = [
-    # drop UCHAR_TYPE override to fix build with system ICU
-    (fetchpatch {
-      url = "https://code.qt.io/cgit/qt/qtwebengine-chromium.git/patch/?id=75f0f4eb";
-      stripLen = 1;
-      extraPrefix = "src/3rdparty/";
-      sha256 = "sha256-3aMcVXJg+v+UbsSO27g6MA6/uVkWUxyQsMD1EzlzXDs=";
-    })
+    ./patches/qtwebengine/0001-task_queue.py-add-jobclient.patch
```

Comment:
Are there plans to upstream these patches? Do we want to maintain these fairly significant changes to the build process?

To be honest I'm not a huge fan of the significant changes and complicated patching.

Reply:
see also https://bugreports.qt.io/browse/QTBUG-95176

what is worse?

* wait for upstream
* overload nix build servers

to review my patches, you would have to review ...

* 300 lines in [jobclient.py](https://github.com/milahu/gnumake-tokenpool/blob/main/py/src/gnumake_tokenpool/jobclient.py) (should be called tokenpool.py)
* 260 lines in [tokenpool.js](https://github.com/milahu/gnumake-tokenpool/blob/main/js/src/gnumake-tokenpool/tokenpool.js)
* [tokenpool-gnu-make.cc](https://github.com/milahu/gnumake-tokenpool/blob/main/cc/src/tokenpool-gnu-make.cc) etc ...  
these are 99% copy-paste from the [tokenpool-master](https://github.com/stefanb2/ninja/tree/topic-tokenpool-master) fork via  [ninja-build/ninja#1139](https://github.com/ninja-build/ninja/issues/1139#issuecomment-461750356)

650 lines of patches

```
$ ( cd pkgs/development/libraries/qt-6/modules/patches/qtwebengine && wc -l * )
  129 0001-task_queue.py-add-jobclient.patch
   24 0002-build_inspector_overlay.py-fix-inherit-fds.patch
  282 0003-gn-add-jobclient.patch
   95 0004-node.py-add-debug.patch
  121 0005-mojom_parser.py-add-jobclient.patch
  651 total
```